### DISCUSSION TITLE
Make cloud based Odoo and Frepple on different domains work together

### DISCUSSION THREAD

**Discussion by etoimene** [frePPLe/frepple#640] [General]
I opened myfrepple.frepple.com instance
I also use Odoo 16 at odoo.myodoo.com
Domains are fictious, so don't try to hack them. :)

I have succesfully installed frepple module in Odoo and configured frepple setting (web token etc.)

I am trying myfrepple.frepple.com/wizard/load/odoo/ at step 2: Configure connection parameters  but keep getting 404.

I found https://frepple.com/docs/current/erp-integration/odoo-connector/installation-and-configuration.php
and it mentions 
"If frePPLe and Odoo are installed on 2 different domains (example: https://myfrepple.frepple.com/ and https://myodoo.odoo.com/), then following lines need to be added:"

I can not modify frepple side obviously. 

Do I need to install frepple locally so that I can change this setting or there is a way around this?

**Comment by etoimene** [comment]
I get only
```
Connection test failed

HTTP Error 404: NOT FOUND
```

on frepple side.

Is there a way to get more detailed message?


Link https://myodoo.odoo.com/frepple/xml?company=My+Company&mode=0&database=TestFrepple

returns 
`Connection OK `
from the browser.

**Comment by jdetaeye** [comment]
>I can not modify frepple side obviously.

The odoo connection wizard is making this change in the frepple backend for you.
![image](https://github.com/frePPLe/frepple/assets/5219207/4b5b551c-509e-46d2-92d2-4f9e97b6e483)

Heads up: We've had some reports where the odoo connection wizard isn't working and/or not returning a good error message when the input parameters aren't working out.     It's on our roadmap to improve this.

If you want the connection to be set up, please send a mail to info@frepple.com and we'll configure the connection manually.

**Comment by etoimene** [comment]
Thanks for input and your kind offer to help.
I have to first doublecheck on my side because I am using nginx with port mapping for multiple Odoo instances.

Did I identify url that frepple uses to connect to odoo correctly? So that I can test ...

**Comment by etoimene** [comment]
It is something on the odoo side, I would say. 
This is what I get in odoo log that came from frepple.
`
2024-03-13 11:41:08,577 3942500 INFO ? werkzeug: 52.16.170.69 - - [13/Mar/2024 11:41:08] "GET /frepple/xml?company=My+Company&mode=0&database=TestFrepple HTTP/1.0" 404 - 1 0.003 0.075`

**Comment by jdetaeye** [comment]
It's in the odoo connector indeed.

You start your debugging journey here  :-) https://github.com/frePPLe/odoo/blob/1a00328fa002b8e8ac96237e956b43a02e162b0b/frepple/controllers/frepplexml.py#L100

We'd be interested to hear your findings. We really want setting up the connection between an odoo instance and frepple smooth and easy.

**Comment by etoimene** [comment]
Just to let you know - I did not drop this, but didn't have time. I'll post my findings.
